docs/reference/contributing/target/itm.md

```diff
@@ -0,0 +1,34 @@
+### Instrumented Trace Macrocell
+
+For targets with Arm CoreSight (for example, Cortex-M3 and Cortex-M4), the Instrumented Trace Macrocell provides a lightweight, nonintrusive way to collect debug trace output. 
+
+#### Assumptions
+
+- The target supports Arm CoreSight.
+- The target has SWO connected either to a compatible interface chip or exposed as a debug pin.
+
+##### Defined behavior
+
+- Targets must implement the function `void itm_init(void)` and add `ITM` to the `device_has` section in `target.json`.
+- When `void itm_init(void)` is called, the debug clock for the ITM must be initialized and the SWO pin configured for debug output.
```

Comment:
Query: Who or what is calling that function? Also, who or what is initializing the debug clock for the ITM and configuring the SWO pin for debug output? You, the user, or something else?